### PR TITLE
new DataSources tutorial

### PR DESCRIPTION
**Pull request by luisaFelixSalles**
This PR adds a new DataSources tutorial.

Sneak peek on how it renders:

image

image

image

### CODE REVIEW

File: doc/source/user_guide/tutorials/data_sources.rst [hunk 1]
```diff
@@ -0,0 +1,785 @@
+.. _ref_tutorials_data_sources:
+
+===============
+DPF DataSources
+===============
```
Comment (PProfizi) [inline doc/source/user_guide/tutorials/data_sources.rst:5]:
```suggestion
===========
DataSources
===========
```

File: doc/source/user_guide/tutorials/data_sources.rst [hunk 2]
```diff
@@ -0,0 +1,785 @@
...
+            from ansys.dpf.core import examples
+
+            # Create the DataSources object
+            my_data_sources_b = dpf.DataSources()
+            # Define the path where the main result data can be found
+            path = examples.find_simple_bar()
+            'C:/Users/user/AppData/local/temp/ASimpleBar.rst'
+            my_data_sources_b.set_result_file_path(filepath=r'C:/Users/user/AppData/local/temp/ASimpleBar.rst', key='rst')
+
+**c) Function** :func:`guess_result_key() <ansys.dpf.core.data_sources.DataSources.guess_result_key>`
```
Comment (PProfizi) [inline doc/source/user_guide/tutorials/data_sources.rst:94]: Hey @jorgepiloto we are seeing annoying errors from Vale which does not exclude links to refs for functions. Do you know how to easily to solve this? Thanks!

Comment (jorgepiloto) [reply doc/source/user_guide/tutorials/data_sources.rst:94]: Let me double check the configuration

Comment (jorgepiloto) [reply doc/source/user_guide/tutorials/data_sources.rst:94]: I am not sure how I was able to accidentally push the commit to `master`. This branch is protected and forcing a pull-request. Moreover, nobody can overcome previous restriction. The commit is this one: https://github.com/ansys/pydpf-core/commit/ca11a17499f35f6a7d8f6be885a27b393aad0589

Comment (jorgepiloto) [reply doc/source/user_guide/tutorials/data_sources.rst:94]: Just cherry-picked in https://github.com/ansys/pydpf-core/pull/1774/commits/912790994a168749a4130134ac667bcb6ca90ae9

File: doc/source/user_guide/tutorials/data_sources.rst [hunk 3]
```diff
@@ -0,0 +1,785 @@
...
+
+Solver output formats supported by DPF
+++++++++++++++++++++++++++++++++++++++
+
+   - MAPDL files: |MAPDL_files|
+   - LS-DYNA files: |LS-DYNA_files|
+   - Fluent files: |Fluent_files1| and |Fluent_files2|
+   - CFX files: |CFX_files1| and |CFX_files2|
+
+This tutoriam will present a general code, with generic files, and examples
```
Comment (luisaFelixSalles) [inline doc/source/user_guide/tutorials/data_sources.rst:26]:
```suggestion
This tutorial will present a general code, with generic files, and examples
```

**Comment by clatapie** [comment]
The ``Vale`` errors should be fixed now. Feel free to reach out if you have any questions!

**Pull request closed without merging**